apis/bigquerybiglake/v1alpha1/table_identity.go

```diff
@@ -92,29 +126,44 @@ func NewTableIdentity(ctx context.Context, reader client.Reader, obj *BigLakeTab
 		if actualParent.Location != location {
 			return nil, fmt.Errorf("spec.location changed, expect %s, got %s", actualParent.Location, location)
 		}
+		if actualParent.CatalogID != catalogID {
+			return nil, fmt.Errorf("spec.catalogRef changed, expect %s, got %s", actualParent.CatalogID, catalogID)
+		}
+		if actualParent.DatabaseID != databaseID {
+			return nil, fmt.Errorf("spec.databaseRef changed, expect %s, got %s", actualParent.DatabaseID, databaseID)
+		}
 		if actualResourceID != resourceID {
 			return nil, fmt.Errorf("cannot reset `metadata.name` or `spec.resourceID` to %s, since it has already assigned to %s",
 				resourceID, actualResourceID)
 		}
 	}
 	return &TableIdentity{
 		parent: &TableParent{
-			ProjectID: projectID,
-			Location:  location,
+			ProjectID:  projectID,
+			Location:   location,
+			CatalogID:  catalogID,
+			DatabaseID: databaseID,
 		},
 		id: resourceID,
 	}, nil
 }
 
 func ParseTableExternal(external string) (parent *TableParent, resourceID string, err error) {
```

Comment:
Suggest moving the parent logic to the parent identity, otherwise we are duplicating the logic among databases, catalogs, and tables. You can look at Apigee for example.

Reply:
Happy to male the template changes and follow that pattern! I used it for ApigeeEnvgroupIntance.